src/herbivoresAndCarnivores.js

```diff
@@ -2,14 +2,38 @@
 
 class Animal {
   // write your code here
+  constructor(name) {
+    this.name = name;
+    this.health = 100;
+    Animal.alive.push(this);
+  }
 }
 
+Animal.alive = [];
+
 class Herbivore extends Animal {
   // write your code here
+  constructor(name) {
+    super(name);
+    this.hidden = false;
+  }
+
+  hide() {
+    this.hidden = !this.hidden;
+  }
 }
 
 class Carnivore extends Animal {
   // write your code here
+  bite(object) {
+    if (object.hidden === false) {
```

Comment:
forgot one condition

>The method does not work if it is another сarnivore, or the herbivore is currently hiding.

```suggestion
    if (!object.hidden) {
```

Reply:
Understood, thx